Wrappers/Python/ccpi/optimisation/operators/FiniteDifferenceOperator.py

```diff
@@ -39,9 +39,17 @@ def __init__(self, domain_geometry,
                        range_geometry=None, 
                        direction=0, 
                        method = 'forward',
-                       bnd_cond = 'Neumann'):
+                       bnd_cond = 'Neumann', **kwargs):
+                        
+        self.label = kwargs.get("label", None)
```

Comment:
Taking direction as a default parameter and label as a kwarg seems strange to me. Especially as if label exists it uses this and ignores direction.

Could you combine them so direction can take the index or the axis name as input?

All you have to do is check if direction is in domain_geometry.dimension_labels as you do, or if it's in range of len(domain_geometry.dimension_labels). 

I also don't think it should have a default value, I think direction should be a required argument.